File: frontend/src/types/graph.ts
export interface FileNode {
  id: string;
  label: string;
  path: string;
  extension: string;
  size: number;
  commit_count: number;
  lines: number;
  node_type: string;
  language: string;
}

export interface ImportEdge {
  id: string;
  source: string;
  target: string;
  weight: number;
}

export interface RepoGraph {
  repo: string;
  nodes: FileNode[];
  edges: ImportEdge[];
  stats: {
    total_files: number;
    total_edges: number;
    languages: Record<string, number>;
    most_changed: Array<{ file: string; path: string; count: number }>;
    max_commits: number;
  };
}

export interface ChatMessage {
  role: "user" | "assistant";
  content: string;
}

File: backend/app/models/graph.py
from pydantic import BaseModel
from typing import Optional

class FileNode(BaseModel):
    id: str
    label: str
    path: str
    extension: str
    size: int
    commit_count: int       # how many times this file changed
    lines: int
    node_type: str          # "file" | "directory"
    language: str

class ImportEdge(BaseModel):
    id: str
    source: str
    target: str
    weight: float           # normalized import strength

class RepoGraph(BaseModel):
    repo: str
    nodes: list[FileNode]
    edges: list[ImportEdge]
    stats: dict

class ChatMessage(BaseModel):
    role: str
    content: str

class ChatRequest(BaseModel):
    messages: list[ChatMessage]
    repo_context: Optional[str] = None
